Fix/update wysiwyg styles

b12f opened the pull request:
https://community.openproject.com/projects/stadt-koeln/work_packages/32602/activity
https://community.openproject.com/projects/stadt-koeln/work_packages/34844/activity

Commit:
Initial class definitions

Commit:
Added more classes

Commit:
Added Table of Contents basics

ulferts commented:
I was surprised to see, that elements like `h1`, `list` and `table` are their own BEM block, leading to classes like `op-uc-list` and `op-uc-h1`. This surprises me especially since the changes also introduce a `op-uc-container`. Before opening the changes I had expected the content to be one block, which would have lead to something like `op-uc` and `op-uc__list`/`op-uc__h1` if I understand BEM correctly. 

I am not saying we should do it in a way I expected it to be done. I just do not understand why it is done in such a way. Can you help me out @b12f ?

b12f commented:
I'm unsure whether the `op-uc-container` is even necessary, which is why I created all the classes independently of that block. Also, I don't like having really big blocks that incorporate a lot of content. You might get weird stuff where there's 5 levels deep of element nesting, instead of multiple blocks, e.g.:

```
.op-uc__blockquote
  .op-uc__table
    .op-uc__table-row
      .op-uc__table-cell
        .op-uc__p
          .op-uc__a
```

even the naming in table here shows that it should really just be a block, imo.

ulferts committed:
op css class for headings

aleixsuau committed:
CkEditor applying custom CSS classes to p, h1, h2, h3, h4, h5, h6, li and blockquote

aleixsuau committed:
CKEditorInspector removed

ulferts committed:
op css class for paragraphs

ulferts committed:
op css class for code/code block

ulferts committed:
adapt specs to altered markdown/html generation

ulferts committed:
adapt grid/budget representers to altered signature

ulferts committed:
op css class for lists

ulferts committed:
op css class for toc

Commit:
Merge branch 'dev' into fix/update-wysiwyg-styles

Commit:
Merge branch 'fix/update-wysiwyg-styles' of github.com:opf/openproject into fix/update-wysiwyg-styles

ulferts committed:
op css class for links

Commit:
Start working on typography css

ulferts committed:
op css class for tables

Commit:
Fixing more typography, trying out larger headers

Commit:
Merge branch 'fix/update-wysiwyg-styles' of github.com:opf/openproject into fix/update-wysiwyg-styles

aleixsuau committed:
Applying custom classes to li, a, blockquote, figure, table, tr, td, th, image, codeblock, figcaption and macros

aleixsuau committed:
Merge branch 'fix/update-wysiwyg-styles' of https://github.com/opf/openproject into fix/update-wysiwyg-styles

ulferts committed:
adapt specs to altered link classes

ulferts committed:
op css class for images

ulferts committed:
apply user content container class throughout application

aleixsuau committed:
CSS alignment custom classes applied to table

aleixsuau committed:
Merge branch 'fix/update-wysiwyg-styles' of https://github.com/opf/openproject into fix/update-wysiwyg-styles

ulferts committed:
op css class for task list checkbox

Commit:
Added task checkbox class

Commit:
Merge branch 'fix/update-wysiwyg-styles' of github.com:opf/openproject into fix/update-wysiwyg-styles

ulferts committed:
amend list checkbox class in backend

ulferts committed:
op css class for table thead element

ulferts committed:
adapt specs on image html generation

Commit:
Updated table and typography styles

Commit:
Update typography and figure styles

Commit:
Figure overflow handling

aleixsuau committed:
Table alignment styles + ckEditor styles removed

aleixsuau committed:
Merge branch 'fix/update-wysiwyg-styles' of https://github.com/opf/openproject into fix/update-wysiwyg-styles

ulferts committed:
rename wiki-anchor to op-uc-link_permalink

ulferts committed:
wrap table in div as well as figure

Commit:
Updated code-block

Commit:
Update permalinks

Commit:
Fixed a lot about tables

Commit:
Removed Description header from work-packages page

Commit:
Fix frontend styles

Commit:
Add placeholder styling, fix toc

Commit:
Fixed figure print

aleixsuau committed:
working with table aligns

aleixsuau committed:
Custom class add to task lists

aleixsuau committed:
Merged with master

aleixsuau committed:
Custom classes applied to theads

ulferts committed:
Merge branch 'dev' into fix/update-wysiwyg-styles

aleixsuau committed:
op-uc-container custom class added to container

ulferts committed:
Merge remote-tracking branch 'origin/fix/update-wysiwyg-styles' into fix/update-wysiwyg-styles

ulferts committed:
Merge remote-tracking branch 'origin/fix/update-wysiwyg-styles' into fix/update-wysiwyg-styles

aleixsuau committed:
Codeblocks inside pre elements

aleixsuau committed:
Merged with origin

aleixsuau committed:
Fix: single <code> and <a> tags

ulferts committed:
explicitly require overwritten gem class

Apparently, the gem is not loaded yet when it is registered as a filter when in eager loading mode

ulferts committed:
adapt spec expectation to altered toc rendering

aleixsuau committed:
CkInspector removed

aleixsuau committed:
Merge branch 'fix/update-wysiwyg-styles' of https://github.com/opf/openproject into fix/update-wysiwyg-styles

Commit:
Latest ckeditor changes

ulferts committed:
remove highlight css class from wiki content

ulferts committed:
allow html pipleline to handle macros with additional classes

Commit:
Fixed a lot of print css for tables

Commit:
Add general print css back in

Commit:
Update Table of Contents styling

aleixsuau committed:
Revert "Custom classes on ul, ol, li and task-lists"

This reverts commit 0d27d281378b324330ea2f25632de898269e2122.

aleixsuau committed:
Custom classes on ul, ol, li and task-lists

aleixsuau committed:
Custom classes on ul, ol, li and task-lists

aleixsuau committed:
Custom classes on ul, ol, li and task-lists

aleixsuau committed:
Custom classes on column's th

ulferts committed:
remove placeholder class when rendering

Commit:
WOrking on task lists

Commit:
Merge branch 'fix/update-wysiwyg-styles' of github.com:opf/openproject into fix/update-wysiwyg-styles

Commit:
Changing task-list classes, changed tests

Commit:
Updated list styles

Commit:
Remove unused todo list styles

ulferts committed:
remove checked in binstubs

Commit:
Fix table of contents

Commit:
Merge branch 'fix/update-wysiwyg-styles' of github.com:opf/openproject into fix/update-wysiwyg-styles

ulferts committed:
adapt todo list handing in backend pipeline

ulferts committed:
adapt specs to altered css classes

Commit:
Add numbers to table of contents

Commit:
Merge branch 'fix/update-wysiwyg-styles' of github.com:opf/openproject into fix/update-wysiwyg-styles

Commit:
Better comments in table of contents

aleixsuau committed:
Fix: wrap single <table> with a <figure>

Commit:
Fixes to todo list design

Commit:
Updated todo list scss to fix nested lists

ulferts committed:
adapt selectors in table spec

Commit:
Update table styles

Commit:
Improve table borders more

aleixsuau committed:
Custom classes specs

aleixsuau committed:
Fix: no need to remove regular list classes when its type changes

Commit:
Add modifier for inline headings

Commit:
Merge branch 'fix/update-wysiwyg-styles' of github.com:opf/openproject into fix/update-wysiwyg-styles

Commit:
Update table editing styles

Commit:
Remove break-word tests

ulferts committed:
wrap images just like tables

Commit:
Update figure content styles

aleixsuau committed:
Fix: All tests passing (ul.op-uc-list_task-list)

aleixsuau committed:
div.op-uc-figure--content wrapping tables

aleixsuau committed:
Specs for figures wrappers div.op-uc-figure--content

aleixsuau committed:
Fix: add custom classes to links and codes again

aleixsuau committed:
Table wrapper div reverted + specs

Commit:
Fix inline palceholders

aleixsuau committed:
Custom macro type classes

Commit:
Add basic macro placeholder changes

ulferts committed:
Merge remote-tracking branch 'origin/dev' into fix/update-wysiwyg-styles

Commit:
Move heading permalink after text

Commit:
Fix word-break spec

aleixsuau committed:
Sending figure styles to the backend (width)

ulferts committed:
extend test to take ckeditor placeholder into account

ulferts committed:
avoid adding bem classes multiple times

ulferts committed:
attempt to fix flickering spec

aleixsuau committed:
Removing image spinner when uploading finishes

ulferts committed:
adapt spec expectations

ulferts reviewed (approved):
Thanks a lot @b12f and @aleixsuau. Looks very good to me.
